### PR TITLE
Fix user creation for edgeAgent and edgeHub

### PR DESCRIPTION
**Pull request by gordonwang0**
Fixes edgeAgent and edgeHub startup scripts, in response to https://github.com/Azure/iotedge/issues/6541
- Creates the edgehubuser and aziot-edged install, just like the edgeagentuser
- Makes edgeagentuser and edgehubuser system accounts with no shell or home folder
- Removes redundant user creation commands
- Fixes typo of edgeHub storage directories

### CODE REVIEW

File: edge-agent/docker/linux/amd64/Dockerfile [hunk 1]
```diff
@@ -8,13 +8,6 @@ ARG EXE_DIR=.
 RUN apk update && \
     apk add --no-cache snappy
 
-# create a default user to allow agent to run as non-root
-# value was chosen as a large value to avoid a typical regular uid, 
-# and matches the "edgeagentuser" created in install packages.
-ARG EDGEAGENTUSER_ID
-ENV EDGEAGENTUSER_ID ${EDGEAGENTUSER_ID:-13622}
-RUN adduser -Ds /bin/sh -u ${EDGEAGENTUSER_ID} edgeagentuser 
-
```
Comment (varunpuranik) [inline edge-agent/docker/linux/amd64/Dockerfile:17]: Why are you removing it from here? Because then the user is not added on host?

Comment (varunpuranik) [reply edge-agent/docker/linux/amd64/Dockerfile:17]: Where does the user get created then?

Comment (varunpuranik) [reply edge-agent/docker/linux/amd64/Dockerfile:17]: I see you moved it to the start script..

Comment (gordonwang0) [reply edge-agent/docker/linux/amd64/Dockerfile:17]: I added it to the install script. The start scripts also do this; there's no need to also create the user here.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh [hunk 2]
```diff
@@ -30,18 +30,11 @@ cuid=$(id -u)
...
 if [ $cuid -eq 0 ]
 then
-
   # Create the agent user id if it does not exist
   if ! getent passwd "${TARGET_UID}" >/dev/null
   then
-    echo "$(date --utc +"%Y-%m-%d %H:%M:%S %:z") Creating UID ${TARGET_UID} as agent${TARGET_UID}"
-    # Use "useradd" if it is available.
-    if command -v useradd >/dev/null
-    then
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh:40]: is this check not needed?

Comment (gordonwang0) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh:40]: Yeah, it's not necessary. The docker containers only contain adduser, so we'll never use useradd. And we exclusively use adduser elsewhere, so I don't think we need to use useradd here.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ then
   if ! getent passwd "${TARGET_UID}" >/dev/null
   then
     echo "$(date --utc +"%Y-%m-%d %H:%M:%S %:z") Creating UID ${TARGET_UID} as edgeagentuser"
-    adduser -D -S -H -s /sbin/nologin -u "${TARGET_UID}" edgeagentuser
+    adduser -D -S -H -s /bin/sh -u "${TARGET_UID}" edgeagentuser
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh]: why this change? Are we sure this won't regress anything?

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh [hunk 4]
```diff
@@ -34,7 +34,7 @@ then
   if ! getent passwd "${TARGET_UID}" >/dev/null
   then
     echo "$(date --utc +"%Y-%m-%d %H:%M:%S %:z") Creating UID ${TARGET_UID} as edgeagentuser"
-    adduser -D -S -H -s /bin/sh -u "${TARGET_UID}" edgeagentuser
```
Comment (varunpuranik) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh]: What does -H flag do? I can't even find it on the man page

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh (hunk 3 above)
Comment (gordonwang0) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh]: This change is just restoring the original value. If you look at the full PR diff, there's no change.

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh (hunk 4 above)
Comment (gordonwang0) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/agentStart.sh]: Specifies that no home directory is created. We do need the home directory for this to work though.

-H is specific to the busybox version of adduser in docker containers, so you won't find it on the manpage for normal adduser.

**Review by varunpuranik — APPROVED** [review]
